examples/tutorials/advanced/cartesian_histograms.py

```diff
@@ -0,0 +1,338 @@
+# using the ``annotate`` parameter.
+
+# Create new figure instance
+fig = pygmt.Figure()
+
+# Create histogram for data01 showing the counts per bin
+fig.histogram(
+    region=[0, 200, 0, len(data01) + 1],
+    projection="X10c",
+    frame=["WSne", "xaf10", "ya5f1+lCounts"],
+    data=data01,
+    series=10,
+    fill="red3",
+    pen="1p,darkgray",
+    histtype=0,
+    # Annotate each bar with the counts it represents
+    annotate=True,
+)
+
+# Shift plot origin 11 centimeters to the right
+fig.shift_origin(xshift="11c")
+
+# Create histogram for data01 showing the cumulative counts
+fig.histogram(
+    region=[0, 200, 0, len(data01) + 1],
+    projection="X10c",
+    frame=["wSnE", "xaf10", "ya5f1+lCumulative counts"],
+    data=data01,
+    series=10,
+    # Use pattern (p) number 8 as fill for the bars
+    # Set the background (+b) to white [Default]
+    # Set the foreground (+f) to black [Default]
+    fill="p8+bwhite+fblack",
+    pen="1p,darkgray",
+    histtype=0,
+    # Show cumulative counts
+    cumulative=True,
+    # Offest (+o) the label by 10 points in negative y-direction
+    annotate="+o-10p",
+)
+
+fig.show()
+
+
+###############################################################################
+# Overlaid bars
+# -------------
+# Overlaid or overlapping bars can be achieved by plotting two or serveral
+# histograms, each for one data set, on top of each other. The legend entry
+# can be specified via the ``label`` parameter.
+#
+# Limitations of histograms with overlaid bars are:
+#
+# - Mixing of colors or/and patterns
+# - Visually more colors or/and patterns than data sets
+# - Visually a "third histogram" (or more in case of more than two data sets)
+
+# Create new figure instance
+fig = pygmt.Figure()
+
+# Create histogram for data01
+fig.histogram(
+    region=[0, 200, 0, 0],
+    projection="X10c",
+    frame=["WSne", "xaf10", "ya1f1+lCounts"],
+    data=data01,
+    series=10,
+    fill="red3",
+    pen="1p,darkgray",
+    histtype=0,
+    # Set legend entry
+    label="data01",
+)
+
+# Create histogram for data02
+# It is plotted on top of the histogram for data01
+fig.histogram(
+    data=data02,
+    series=10,
+    # Fill bars with color "orange", use a transparency of 50% ("@50")
+    fill="orange@50",
+    pen="1p,darkgray",
+    histtype=0,
+    label="data02",
+)
+
+# Add legend
+fig.legend()
+
+fig.show()
+
+
+###############################################################################
+# Stacked bars
+# ------------
+# Stacked bars can be achieved similar to overlaid bars via plotting two or
+# several histograms on top of each other. However, before plotting, combined
+# data sets have to be created from the singel data sets.
+#
+# Limitations of histograms with stacked bars are:
+#
+# - No common baseline
+# - Partly not directly clear whether overlaid or stacked bars
+
+# Combine the two data sets to one data set
+data_merge = np.concatenate((data01, data02), axis=None)
```

Comment:
I'm a little confused about "Stacked Bars". It usually makes no sense to combine two datasets into a single one and then plot the histogram for the combined dataset. It's also more confusing that the combined dataset is labeled as "data02" in the legend.

I think we should remove the "Stacked bars" section.

Reply:
> I'm a little confused about "Stacked Bars". 

Yes, I know this is an "interesting" and not straightforward way to create a histogram with stacked bars.

>  It usually makes no sense to combine two datasets into a single one and then plot the histogram for the combined dataset. 

Yes. Unfortunately, so far I found no better way to consider the counts of dataset 01 to have the bars for dataset02 appear on top of the bars for dataset01.
@GenericMappingTools/pygmt-maintainers are there maybe some better ideas how to accomplish this, please 🙂?

> It's also more confusing that the combined dataset is labeled as "data02" in the legend.

"data02" is added here as a label for the combined dataset, as in the final histogram the combined dataset appears visually as "data02" in the context of stacked bars. 

> I think we should remove the "Stacked bars" section.

I added this section mainly for the sake of completeness (overlaid, stacked, grouped bars). I am fine with removing it (or even better, replacing the code in case there is a better idea).

Reply:
Sorry that I misunderstood the meaning of "Stacked bars" as I have never used it. After searching for more examples of "stacked bars" (e.g., https://stats.stackexchange.com/questions/407195/when-to-stack-histograms), now the figure makes sense to me. I think we just need to make it clear that: PyGMT doesn't have direct support for stacked histograms, but we can still plot stacked histograms by plotting the "combined" datasets.

Reply:
What about adding one more type of histogram by setting `stairs=True` and using different pen attributes?

Similar to the left-bottom panel in this figure (https://matplotlib.org/stable/gallery/statistics/histogram_multihist.html):
![image](https://user-images.githubusercontent.com/3974108/235304745-c9ddc3e1-bed1-49ce-aafa-0113acba770d.png)

Reply:
I have added the section "stair-steps" as second section of this tutorial.

Reply:
> I think we just need to make it clear that: PyGMT doesn't have direct support for stacked histograms, but we can still plot stacked histograms by plotting the "combined" datasets.

I updated the comments / added some explanation in commit https://github.com/GenericMappingTools/pygmt/pull/2445/commits/4e69442a3c4e834d918e9d6ecbf9d5c6afdb46fe.